drivers/stepper_motor_controller/Kconfig

```diff
@@ -0,0 +1,48 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 Carl Zeiss Meditec AG
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig STEPPER_MOTOR_CONTROLLER
+	bool "Stepper Motor Controller"
+	help
+		Enable stepper motor controller
+
+if STEPPER_MOTOR_CONTROLLER
+
+module = STEPPER_MOTOR_CONTROLLER
+module-str = stepper_motor_controller
+source "subsys/logging/Kconfig.template.log_config"
+
+config STEPPER_MOTOR_CONTROLLER_INIT_PRIORITY
+	int "Stepper Motor Controller init priority"
+	default 90
+	help
+	  Stepper motor controller initialization priority.
+
+config STEPPER_MOTOR_CONTROLLER_SHELL
+	bool "Shell for stepper motor controllers"
+	depends on SHELL
+	default y if !SHELL_MINIMAL
+	help
+	  This shell provides specialised access to basic stepper motor controller shell
+	  to execute various functions.
+
+config STEPPER_MOTOR_CONTROLLER_HAS_CONSTANT_VELOCITY_MODE
```

Comment:
I would assume that these settings are more like a "per controller" setting. If they are KConfig parameters these would be global instead.

Would there be a more elegant solution to have this as "per controller" setting?

Reply:
I know its elegant. Probably some help will come in from the community :-)